File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(int i=0;i<s.length();i++){
            char ch=s[i];
            if(ch=='{'||ch=='['||ch=='('){
                st.push(ch);
            }else{
                if(!st.empty() && abs(ch-st.top())<=2){
                    st.pop();
                }
                else{
                    return false;
                }    
            }        
        }
        return st.empty();
        
    }
};

File: Data Structures & Algorithms/simplify-path/submission-1.java
class Solution {
    public String simplifyPath(String path) {
        String strings[] = path.split("/");
        Stack<String> stack = new Stack<>();


        for(String str: strings){

            if(str.equals(".")||str.equals("")){
                continue;
            }
            else if(str.equals("..")){
                if(!stack.isEmpty()){
                    stack.pop();
                }
            }else{
                stack.push(str);
            }
        }

        StringBuilder res = new StringBuilder();

        for(String s: stack){
            res.append("/").append(s);
        }
        if(res.length() == 0){
            return "/";
        }
        return res.toString();

    }
}